justfile

```diff
@@ -0,0 +1,42 @@
+set dotenv-load
+set ignore-comments
+
+# Default recipe
+[private]
+default:
+    @just --list
+
+# Install necessary tools
+install-tools:
+    rustup component add rustfmt --toolchain nightly
+    cargo install cargo-sort
+
+# Check if necessary tools are installed
+[private]
+check-tools:
+    #!/usr/bin/env bash
+    if ! cargo +nightly fmt --help &> /dev/null; then 
+        echo "cargo +nightly fmt is not installed, use just install-tools or install it manually"
+        exit 1
+    fi
+    if ! cargo sort --help &> /dev/null; then 
+        echo "cargo sort is not installed, use just install-tools or install it manually"
+        exit 1
+    fi
+
+# Format code
+format: check-tools
+    cargo clippy --all-targets --all-features
+    cargo +nightly fmt -- --check
+    cargo sort -w . --check
```

Comment:
I would not include clippy as a step for `just format` that I expect to have only cosmetic checks

This is more related to `just check` for which I would include both cargo check and clippy.